File: metodo_tree.cpp
#include<iostream>
using namespace std;
// metodo de ordenamiento arbol binario de busqueda c++
struct Node {
    int data;
    Node* left;
    Node* right;
};
Node* newNode(int data) {
    Node* node = new Node();
    node->data = data;
    node->left = nullptr;
    node->right = nullptr;
    return node;
}
Node* insert(Node* root, int data) {
    if (root == nullptr) {
        return newNode(data);
    }
    if (data < root->data) {
        root->left = insert(root->left, data);
    } else {
        root->right = insert(root->right, data);
    }
    return root;
}
void inorder(Node* root) {
    if (root != nullptr) {
        inorder(root->left);
        cout << root->data << " ";
        inorder(root->right);
    }
}
int main() {
    Node* root = nullptr;
    int arr[] = {5, 3, 8, 1, 4, 7, 9};
    cout << "Elementos a ordenar: ";
    for (int i = 0; i < sizeof(arr) / sizeof(arr[0]); i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
    
    int n = sizeof(arr) / sizeof(arr[0]);
    for (int i = 0; i < n; i++) {
        root = insert(root, arr[i]);
    }
    cout << "Elementos ordenados: ";
    inorder(root);
    cout << endl;
    return 0;
}

File: metodo_tree.py
# metodo de ordenamiento tree sort
class Node:
    def __init__(self, key):
        self.left = None
        self.right = None
        self.val = key
def insert(root, key):
    if root is None:
        return Node(key)
    else:
        if root.val < key:
            root.right = insert(root.right, key)
        else:
            root.left = insert(root.left, key)
    return root
def inorder_traversal(root, res):
    if root:
        inorder_traversal(root.left, res)
        res.append(root.val)
        inorder_traversal(root.right, res)
    return res

# Ejemplo de uso
numeros = [5, 3, 8, 1, 4]
r = Node(numeros[0])
for num in numeros[1:]:
    insert(r, num)
sorted_numeros = inorder_traversal(r, [])
print("Arreglo original:", numeros)
print("Arreglo ordenado:", sorted_numeros)